assembl/models/attachment.py

```diff
@@ -172,6 +183,33 @@ def __init__(self, *args, **kwargs):
 
     # Should we defer this?
     data = Column(LargeBinary, nullable=False)
+    # Note: use SQLA 1.1 for a better Enum behaviour
+    av_checked = Column(Enum(*AntiVirusStatus.__members__.keys(),
+                             name="anti_virus_status"),
+                        server_default='unchecked')
+
+    def guess_extension(self):
+        extensions = guess_all_extensions(self.mime_type)
+        if extensions:
+            # somewhat random
+            return extensions[0]
+
+    def check_for_viruses(self, antivirus=None):
+        "Check if the file has viruses"
+        antivirus = antivirus or get_antivirus()
+        safe = antivirus.check(self.data, self.guess_extension())
+        status = AntiVirusStatus.passed.name if safe else AntiVirusStatus.failed.name
+        self.av_checked = status
+        return status
+
+    def safe_data(self):
+        if self.av_checked == AntiVirusStatus.unchecked.name:
+            needs_check = self.discussion.preferences['requires_virus_check']
+            if needs_check:
+                self.av_checked = self.check_for_viruses()
+        if self.av_checked == AntiVirusStatus.failed.name:
+            return ''
```

Comment:
I think as a truly base implementation, this may suffice, but it can certainly improved upon. We could have an image placeholder for AV failed files, and instead of returning `''`, you can return a document object representing this placeholder. @paolina92 Can ask Roxana to make a quick image for AV failed if needed quickly.

Reply:
yes as well as this function probably needs to return immediately instead of wait until av check is done

Reply:
So I think we need to add a "check_in_progress" state or something, so that on first access to the file, status is "unchecked", and we add the check in a job and set status to "check_in_progress" and return a placeholder. (See how other web services like gmail do it)